.changeset/perfect-jobs-remember.md

```diff
@@ -0,0 +1,6 @@
+---
+'@crowdstrike/glide-core': minor
+---
+
+- Checkbox Group's `value` is no longer reflected to match native.
```

Comment:
With native form controls, `value` generally isn't reflected so it can be reverted back to its original value when the form is reset. 

However, some controls (like `<input type="checkbox">` and `<option>`)  do reflect their `value` because the user can't change `value` by interacting with the control

But Checkbox Group is different than a single, native checkbox because the user can change its `value` by checking and unchecked checkboxes.